### PR TITLE
Add Go overrides files

### PR DESCRIPTION
**Pull request by saicheems**
Also update pubsub overrides for Node.js

### CODE REVIEW

**Comment by saicheems** [comment]
This is actually wrong - gimme a bit to fix it

**Comment by saicheems** [comment]
Updated

Btw, it strikes me that the Go client library is wrong for these cases - these methods are page streaming, but the response doesn't contain the field `nextPageToken`. Java seems to handle these APIs correctly, but since there's not `nextPageToken` field (only `pageToken`), Go doesn't generate the necessary boilerplate for page streaming. I wonder if this is something that we should file against `google-api-go-client`.

**Comment by saicheems** [comment]
ping

File: src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline [hunk 1]
```diff
@@ -67,7 +67,7 @@ authorize(function(authClient) {
   var request = {
     // REQUIRED: The resource for which the policy is being specified. `resource` is usually specified as
     // a path. For example, a Project resource is specified as `projects/{project}`.
-    resource: '',  // TODO: Update placeholder value.
+    resource_: '',  // TODO: Update placeholder value.
```
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline]: This is what the client library expects, cloudresourcemanager has a field `resource`, and the Node.js library maps requestBody to the `resource` parameter. To resolve this conflict, the library owners have decided to rename any parameters called `resource` to `resource_`.

So, this is a deliberate design choice in the client libraries, so this is correct.

File: src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline [hunk 2]
```diff
@@ -18,7 +18,7 @@ authorize(function(authClient) {
   var request = {
     // REQUIRED: The resource for which the policy is being requested. `resource` is usually specified as
     // a path. For example, a Project resource is specified as `projects/{project}`.
-    resource: '',  // TODO: Update placeholder value.
+    resource_: '',  // TODO: Update placeholder value.
```
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline]: it does:
https://github.com/google/google-api-nodejs-client/blob/master/apis/pubsub/v1.js#L1762

File: src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline (hunk 1 above)
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline]: https://github.com/google/google-api-nodejs-client/blob/master/apis/pubsub/v1.js#L1726

Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_pubsub.v1.json.baseline]: OK

**Comment by vchudnov-g** [comment]
If compile checks pass, LGTM.

Do file that issue and maybe link to it in this PR as well.

**Comment by saicheems** [comment]
https://github.com/google/google-api-go-client/issues/176

**Pull request merged**